lib/buffer.js

```diff
@@ -1255,35 +1253,41 @@ function btoa(input) {
     throw new ERR_MISSING_ARGS('input');
   }
   input = `${input}`;
+  let acc = 0;
   for (let n = 0; n < input.length; n++) {
-    if (input[n].charCodeAt(0) > 0xff)
-      throw lazyDOMException('Invalid character', 'InvalidCharacterError');
+    acc |= StringPrototypeCharCodeAt(input, n);
+  }
+  if (acc & ~0xff) {
+    throw lazyDOMException('Invalid character', 'InvalidCharacterError');
   }
   const buf = Buffer.from(input, 'latin1');
   return buf.toString('base64');
 }
 
 // Refs: https://infra.spec.whatwg.org/#forgiving-base64-decode
+//       https://infra.spec.whatwg.org/#ascii-whitespace
+// Valid Characters: [\t\n\f\r +/0-9=A-Za-z]
+// Lookup table (-1 = invalid, 0 = valid)
```

Comment:
Why did you select -1 and 0, instead of 0 and 1?

Reply:
See my comment for line 1308.